tools/autograd/templates/python_variable_methods.cpp

```
@@ -822,6 +1015,14 @@ static PyObject * THPVariable_type(PyObject* self, PyObject* args, PyObject* kwa
 ${py_methods}
 
 static PyObject * THPVariable_bool_scalar(PyObject* self, PyObject* args) {
+  if (check_has_torch_function(self)) {
+    try {
+      return handle_torch_function(self, "__bool__");
+    }
+    catch(const python_error&) {
```

Are you sure you want to catch every python error and swallow all of them?

---

A: It isn't swallowing

---

Better to use the macros directly